LCD.c

```diff
@@ -0,0 +1,208 @@
+
+    letter = letter << 4;                   //analyze next 4 bits
+    P4 -> OUT = letter;                     //shifted letter data on OUT
+    nybble();
+
+    delay_ms(2,FREQ_48_MHZ);
+}
+
+void write_string_LCD(char* string)
+{
+    while(*string)                          //loop until string completed
+    {
+        write_char_LCD(*string);            //write to LCD
+        string++;                           //count "up" to next char in string
+    }
+}
+
+void SetCursorLocation(char place)          //set cursor to DDRAM location as defined
+{                                           //by the STU data sheet for 2-line mode
+    char myCommand = ENABLE_MOVE + place;
+    command_LCD(myCommand);
+}
+
+void write_LCD_L1(char* string)
+{
+    SetCursorLocation(0x00);                //set cursor location to DDRAM address 0x00, line 1
+    write_string_LCD(string);               //string to be assigned to line 1
+}
+
+void write_LCD_L2(char* string)
+{
+    SetCursorLocation(0x40);                //set cursor location to DDRAM address 0x40, line 2
+    write_string_LCD(string);               //string to be assigned to line 2
+}
+
+void write_LCD_L3(char* string)
+{
+    SetCursorLocation(0x14);                //set cursor location to DDRAM address 0x14, line 3
+    write_string_LCD(string);               //string to be assigned to line 3
+}
+
+void write_LCD_L4(char* string)
+{
+    SetCursorLocation(0x54);                //set cursor location to DDRAM address 0x54, line 4
+    write_string_LCD(string);               //string to be assigned to line 4
+}
+
+void nybble(void)
+{
+    P3-> OUT |= EN;                         //enable high
+    delay_ms(1, FREQ_48_MHZ);               //delay 0.1ms
+    P3-> OUT &= ~EN;                        //enable low
+}
+
+void StartScreen(void)                      //STARTING SCREEN
+{
+    write_LCD_L1("Startup             ");
+    write_LCD_L2("Enter Date:__/__/__ ");   //remember: only have 20 char spaces
+    write_LCD_L3("Enter Time:__:__    ");   //military time
+    write_LCD_L4("*Clear        #Enter");
+    SetCursorLocation(0x4B);
+    //getkeypress function
+    //replace "_" with keypress
+    //once key is pressed, will cursor automatically move right, or do we need to define next location?
+    //needs to wait for either Clear or Enter press before changing screens
+}
+
+void HomeScreen(void)                       //HOME MODE CHOICE SCREEN
+{
+    write_LCD_L1("A: Manual Entry     ");   //A, B, or C key press will decide next LCD screen
+    write_LCD_L2("B: Algorithm Based  ");
+    write_LCD_L3("C: Demo Mode        ");
+    write_LCD_L4("*Back               ");
+}
+
+void A1_MANUAL(void)                        //A_1: MANUAL ENTRY INPUT SCREEN
+{
+    write_LCD_L1("                    ");
+    write_LCD_L2("Enter Angle: ___deg ");
+    write_LCD_L3("                    ");
+    write_LCD_L4("*Back  !Home  #Enter");   //Back deletes input value, Home goes Home, Enter verifies angle desired
```

Comment:
we don't have '!' on keypad... maybe we could use the '.' since the angles are only whole numbers?

Reply:
Oh right, changed all "!" to "."